dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaManagerConnector.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+
+@AutoService({Connector.class})
+@Description("Dumps metadata from Cloudera Manager.")
+@RespectsInput(
+    order = 100,
+    arg = ConnectorArguments.OPT_URI,
+    description = "The Cloudera's Manager API HTTP(s) endpoint.")
+@RespectsInput(
+    order = 200,
+    arg = ConnectorArguments.OPT_USER,
+    description = "The username for Cloudera's Manager.")
+@RespectsInput(
+    order = 300,
+    arg = ConnectorArguments.OPT_PASSWORD,
+    description = "The password for Cloudera's Manager.",
+    required = "If not specified as an argument, will use a secure prompt")
+@RespectsInput(
+    order = 400,
+    arg = ConnectorArguments.OPT_CLUSTER,
+    description = "The name of Cloudera's cluster.",
+    required = "Only if you need to dump data for a single Cloudera cluster")
+public class ClouderaManagerConnector extends AbstractConnector {
+
+  public static final String FORMAT_NAME = "cloudera-manager.dump.zip";
+
+  public ClouderaManagerConnector() {
+    super("cloudera-manager");
+  }
+
+  @Nonnull
+  @Override
+  public String getDefaultFileName(boolean isAssessment, Clock clock) {
+    return ArchiveNameUtil.getFileNameWithTimestamp(getName(), clock);
+  }
+
+  @Override
+  public void addTasksTo(@Nonnull List<? super Task<?>> out, @Nonnull ConnectorArguments arguments)
+      throws Exception {
+    out.add(new DumpMetadataTask(arguments, FORMAT_NAME));
+    out.add(new FormatTask(FORMAT_NAME));
+    out.add(new ClouderaClustersTask());
+    out.add(new ClouderaCMFHostsTask());
+    out.add(new ClouderaAPIHostsTask());
+    out.add(new ClouderaServicesTask());
+  }
+
+  @Nonnull
+  @Override
+  public ClouderaManagerHandle open(@Nonnull ConnectorArguments arguments) throws Exception {
+    List<String> errors = new ArrayList<>();
+    if (arguments.getUri() == null) {
+      errors.add("--url for Cloudera Manager API is required");
+    }
+    if (arguments.getUser() == null) {
+      errors.add("--user is required for Cloudera Manager API connector");
+    }
+
+    if (!errors.isEmpty()) {
+      throw new MetadataDumperUsageException(
+          "Missing arguments for connector generic-args : ", errors);
+    }
+    URI uri = new URI(arguments.getUri());
+
+    CloseableHttpClient httpClient = disableSSLVerification(HttpClients.custom()).build();
+    ClouderaManagerHandle handle = new ClouderaManagerHandle(uri, httpClient);
+
+    String user = arguments.getUser();
+    String password = arguments.getPasswordOrPrompt();
+    doClouderaManagerLogin(handle.getBaseURI(), httpClient, user, password);
+
+    return handle;
+  }
+
+  private void doClouderaManagerLogin(
+      URI baseURI, CloseableHttpClient httpClient, String user, String password) throws Exception {
+    ClouderaManagerLoginHelper.login(baseURI, httpClient, user, password);
+  }
```

Comment:
Helper method called from only one location - should it inline?

Reply:
Initially it was private method in this class, then I decided to extract it into separate class for testing simplifications.